src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,172 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value, null, tail);
```

Comment:
When creating a new `Node<T>`, pass all the fields to the `Node` constructor. However, you should also consider refactoring the constructor to avoid directly accessing fields of the `Node` class from outside, as `Node` should be an inner class.